.github/workflows/scorecards.yml

```diff
@@ -0,0 +1,69 @@
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '45 9 * * 3'
+  push:
+    branches: [ "main" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

Comment:
Is there any reason not to use `actions/checkout@v4`?

Reply:
Yeah, let's not ping specific shas here but the latest semantic version.

Reply:
IMO I think we should not change this as in the scorecard action there is a check for [pinned dependencies](https://github.com/ossf/scorecard/blob/cfe5c5f9c3dbf2a98c864e1b1ea25fae289f3b9d/docs/checks.md#pinned-dependencies) (this project has many workflows which are not pinned by hash resulting score of 0 which needs improvement) so if we change this it will further drop the score.

Reply:
We should not pin dependencies imho without also added automation to update it. Since otherwise it never gets updated.

Reply:
I assumed that the project already has dependabot, Sorry for that, I will make the changes. Also I suggest adding Dependency-Update-Tool to the project; it will be a good addition, as [Dependency-Update-Tool](https://github.com/ossf/scorecard/blob/a8eae2d833553e93021a22f5641661cb6ad3e24c/docs/checks.md#dependency-update-tool) is also one of the scorecard checks, and most of the CNCF projects use it. With the addition of the dependency update tool, we will be able to pin dependencies, which will improve the scorecard score.

Reply:
We do have dependabot running on the project. However, I have not seen it update GH workflows, and in some other context we found that bots are not allowed to update GH workflows.

Reply:
bot PRs: https://github.com/vitessio/vitess/pulls?q=is%3Apr+author%3Aapp%2Fdependabot+is%3Aclosed